src/components/grid-list/grid-list.ts

```diff
@@ -1,8 +1,127 @@
-import {Component} from '@angular/core';
+import {
+  Component,
+  ViewEncapsulation,
+  AfterContentChecked,
+  OnInit,
+  Input,
+  ContentChildren,
+  QueryList,
+  Renderer,
+  ElementRef
+} from '@angular/core';
+import {MdGridTile} from './grid-tile';
+import {TileCoordinator} from './tile-coordinator';
+import {
+    TileStyler,
+    RowHeightMode,
+    FitTileStyler,
+    RatioTileStyler,
+    FixedTileStyler
+} from './tile-styler';
+import {Dir} from '../../core/rtl/dir';
+
+// TODO(kara): Conditional (responsive) column count / row size.
+// TODO(kara): Re-layout on window resize / media change (debounced).
+// TODO(kara): gridTileHeader and gridTileFooter.
 
 @Component({
   selector: 'md-grid-list',
+  host: { 'role': 'list' },
   templateUrl: './components/grid-list/grid-list.html',
   styleUrls: ['./components/grid-list/grid-list.css'],
+  encapsulation: ViewEncapsulation.None
 })
-export class MdGridList {}
+export class MdGridList implements OnInit, AfterContentChecked {
+  /** Number of columns being rendered. */
+  _cols: number;
+
+  /** Row height value passed in by user */
+  _rowHeight: string;
+
+  /** The amount of space between tiles. This will be something like '5px' or '2em'. */
+  _gutter: string = '1px';
+
+  /** Sets position and size styles for a tile */
+  _tileStyler: TileStyler;
+
+  /* Array of grid tiles being rendered (from query list) */
+  _tiles: MdGridTile[] = [];
+
+  constructor(private _renderer: Renderer, private _element: ElementRef,
+              private _dir: Dir) {}
+
+  get cols() { return this._cols; }
+
+  @Input('cols')
```

Comment:
You don't need the `'cols'` string if it matches the property name. 

Also, I think we've generally been putting the `@Input` on the getter rather than the setter (it should be identical AFAIK).